tools/build/rules/grpc_c++.bzl

```diff
@@ -110,6 +110,156 @@ def grpc_cc_library(name, srcs = [], public_hdrs = [], hdrs = [],
     alwayslink = alwayslink,
   )
 
+
+"""Generates C++ grpc stubs from proto_library rules.
+
+This is an internal rule used by cc_grpc_library, and shouldn't be used
+directly.
+"""
+
+def generate_cc_impl(ctx):
```

Comment:
Names that start with an underscore are considered private. So if you don't want functions to be used outside of this file, name them `_foo`. This is typically always done for rule implementations.

Reply:
Done in 4caa404